Feature/add elevator

sphillip2021 opened the pull request:
Working on commands and subsystems for the elevator(don't have stuff for the encoders yet).

Commit:
Added simple PCM subsystem

smatsumo2019 committed:
Created Claw Subsystem

Commit:
Removed commented out example code

sphillip2021 committed:
Merge pull request #7 from NorthernForce/feature/code_cleanup

removed unused classes

Commit:
added commands and subsystem for cargo manipulator

smatsumo2019 committed:
Created Command for HatchClaw

Made constants in RobotMap as well to make code cleaner

Commit:
Added basic current limiting (needs to be tuned)

Commit:
Add speed divider so it will start fast and then go slower.

smatsumo2019 committed:
Added Commands for Claw Subsystem

Commit:
Using timer class for speed divider instead of counter in periodic.

sphillip2021 committed:
Merge pull request #9 from NorthernForce/test/rebasepcm

Test/rebasepcm

sphillip2021 committed:
Mistakes are here

sphillip2021 committed:
merge

smatsumo2019 committed:
Uncommented Claw Code

Uncommented Claw Code so we can test it and fix it accordingly.

smatsumo2019 committed:
Renamed Claw Commands Appropriately

Changed ExtendClaw to LowerClaw
Changed RetractClaw to RaiseClaw

sphillip2021 committed:
Elevator Commands and Subsystems

smatsumo2019 committed:
Added PCM ID to Solenoids

Commit:
This should work.

Commit:
ah

snutt committed:
Configure SparkMax controllers

snutt committed:
Update BrushlessDrive.cpp

Commit:
Merge pull request #12 from NorthernForce/feature/Add_drive_motor_limits

Configure SparkMax controllers

lgreenlaw2022 committed:
Added positions and put commands on dashboard

lgreenlaw2022 committed:
more encoder stuff

Commit:
Need one small fix

Commit:
Shuffleboard test added

davewood-maine committed:
add log statements for debugging

davewood-maine committed:
more debug

lgreenlaw2022 committed:
added debug code

Commit:
merged master into branch

lgreenlaw2022 committed:
changed motion magic and pidIdx

Commit:
Need updates, had to change a depreciated function.

Commit:
Set mode to brake.

Commit:
Merge remote-tracking branch 'origin/master' into feature/add_cargo_manipulator

I need to figure out how to rebase.

Commit:
Must change ALL to brake.

Commit:
Merge remote-tracking branch 'origin/master' into feature/add_cargo_manipulator

this is not practical.

Commit:
This now works

davewood-maine committed:
more debug for motion magic

smatsumo2019 committed:
Merge pull request #13 from NorthernForce/feature/add_cargo_manipulator

Feature/add cargo manipulator

davewood-maine committed:
debug

gaudetja commented on DeepSpace/src/main/cpp/Robot.cpp:
```diff
@@ -7,13 +7,18 @@
+//std::shared_ptr<Claw> Robot::m_claw;
 
 void Robot::RobotInit() {
   std::cout << "RobotInit Started" << std::endl;
```
please replace std:endl with "\n"

gaudetja commented on DeepSpace/src/main/cpp/Robot.cpp:
```diff
@@ -36,7 +52,17 @@ void Robot::RobotInit() {
+  std::cout << "Elevator current position: " << pos 
+    << ", motor output: " << motorOutput
+    << ", sensor velocity: " << velocity
+    << ", error " << err << std::endl;
```
please replace std:endl with "\n"

gaudetja commented on DeepSpace/src/main/cpp/Robot.cpp:
```diff
@@ -62,30 +88,12 @@ void Robot::DisabledPeriodic() { frc::Scheduler::GetInstance()->Run(); }
  */
 void Robot::AutonomousInit() {
   std::cout << "AutonomousInit Started" << std::endl;
```
please replace std:endl with "\n"

gaudetja commented on DeepSpace/src/main/cpp/commands/ElevatorSetPosition.cpp:
```diff
@@ -0,0 +1,70 @@
+  m_elevator(Robot::m_elevator.get())
+{
+  
+  std::cout << "Created elevator set position with position " << std::endl;
```
please replace std:endl with "\n"

gaudetja committed:
use position instead of motionprofile

jcrosbyME committed:
BrushlessDrive: Added Open Loop Ramp

lgreenlaw2022 committed:
pid tuning

lgreenlaw2022 committed:
merge from master

lgreenlaw2022 committed:
fixed merge

lgreenlaw2022 committed:
limit switches and more pid tuning

jcrosbyME committed:
Added Ultrasonic Range Finder Subsystem

smatsumo2019 committed:
Added States for toggle command

smatsumo2019 committed:
getDistance gets voltage and divides now

Used to return 0 but now it should take the analog input from the sensor and divide it by 0.0098 (9.8mv/Inch) to get distance in inches to whatever is in front of the robot.

smatsumo2019 committed:
Gave ToggleClawRaise some meaning

ToggleClawRaise now takes the state of the claw and determines whether it should raise or close every time. Also now it is a timedcommand like raise and lower claw (like it should've been)

jcrosbyME committed:
Added PIDSource to RangeFinder

smatsumo2019 committed:
Added command with button in OI

smatsumo2019 committed:
Limited Cargo Intake Wheel's Currents

smatsumo2019 committed:
Removed extra semicolon

smatsumo2019 committed:
Added the trigger button command assignment

The commands should be assigned to the right buttons for intaking cargo, ejecting cargo, and opening and closing the claw.

Commit:
Merge pull request #15 from NorthernForce/feature/set_cargo_manipulator_current_limits

Limited Cargo Intake Wheel's Currents

Commit:
Merge pull request #16 from NorthernForce/feature/add_clawsolenoid_controller

Feature/add clawsolenoid controller

lgreenlaw2022 committed:
added robot setup commands and buttons on shuffleboard

smatsumo2019 committed:
Elevator extends and retracts

Added commands and solenoid in elevator for extending the elevator. Did not implement in teleopinit and stuff.

lgreenlaw2022 committed:
put buttons on dashboard and changed motor speed

jcrosbyME committed:
Changed Encoder Values to new Estimates

jcrosbyME committed:
Multiple Updates Merged Development

jcrosbyME committed:
Added Light Ring Subsystem

jcrosbyME committed:
LIght Ring Addition

lgreenlaw2022 committed:
elevator tuning

lgreenlaw2022 committed:
updated positions and butttons - untested

gaudetja committed:
Merge branch 'robot_testing' into feature/add_elevator
